sstables/size_tiered_backlog_tracker.hh

```diff
@@ -89,7 +89,7 @@ class size_tiered_backlog_tracker final : public compaction_backlog_tracker::imp
     inflight_component compacted_backlog(const compaction_backlog_tracker::ongoing_compactions& ongoing_compactions) const;
 
     double log4(double x) const {
-        static constexpr double inv_log_4 = 1.0f / std::log(4);
+        double inv_log_4 = 1.0f / std::log(4);
```

Comment:
I think this change is ok, but it makes me feel uneasy:
I think if the compiler knows how to calculate / optimize this expression during compile time, that's the quintessential "constexpr", and actually writing constexpr was a nice way to emphasize that you insist that this optimization must be done, so it's sad clang doesn't allow us to use it. Additionally, the fact you removed the "static" means you count even more on this optimization being done (because if it's not, better do the actual calculation only once! that's what the "static" does). I hope you are right that it is. If we had a reason to suspect it isn't, keeping the "static" would have made sense.

Reply:
constexpr is not about optimization. The compiler is free to emit runtime code to generate it. constexpr is about compile-time values, but it only forces compile-time evaluation if the value is used in a compile-time context.

Reply:
I think it's clear that the intention of the author was to somehow guarantee that this 1/log(4) will not be calculated again and again on every call, and hence the use of "static" to make sure it's only calculated once, and "constexpr" as "fail this if it can't calculate at compile time". Your fix is ok - but it means that we just trust the optimizer to optimize this in any case - without all the tricks that the author tried to use. As optimizers are getting better (and slower), I guess we can just trust this...